### DISCUSSION TITLE
`object` method inside an `Alba::Resource`

### DISCUSSION THREAD

**Discussion by davidgm0** [okuramasafumi/alba#319] [Q&A] [answered]
I'm new to Alba and I come from AMS. While trying to migrate, I was trying to serialize a list of objects:

`SomeSerializer.new([user1, user2]).serialize`

In my serializer I had:

```ruby
class SomeSerializer
  include Alba::Resource # This was before an `ActiveModel::Serializer`

  attribute :id do
    object.token
  end
end
```
This was giving me an error: 

```ruby
*** NoMethodError Exception: undefined method `token' for #<ActiveRecord::AssociationRelation [#<User... 
```

This is because object for `Alba` is the 'whole array' passed, where for AMS is the current object being serialized.

I had to switch to

```ruby
attribute :id do |resource|
    resource.token
end
```

This will require substantial changes to the serializers. What's the background on the whole object being available to the serializer? I was thinking if I could define somehow an `object` method that gets the resource, so I don't need to do some changes, but I don't think that's going to work.

**Comment by okuramasafumi** [comment, marked as answer]
@davidgm0 Actually there IS `object` method, but it points to the object given to a resource class during initialization.

```ruby
# this `users` is `object` in serializer class
SomeSerializer.new(users).serialize
```

Alba is not meant be 100% compatible with AMS, but we'd like to make migration as easy as possible.
In this case, if `object` is `user` instead of `users`, it's better for migration.

One simple way is to rename a block argument from `resource` to `object`.

```ruby
# This should work
attribute :id do |object|
    object.token
end
```

Then it's easier to do so with automation such as editor macro, just adding ` |object|` to each end of `attribute` line.

**Comment by davidgm0** [comment]
Good tip, thanks!
